aspnetcore/web-api/advanced/formatting/samples/6.x/ResponseFormattingSample/Snippets/Program.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Microsoft.AspNetCore.Mvc.Formatters;
+using Newtonsoft.Json.Serialization;
+
+namespace ResponseFormattingSample.Snippets;
+
+public static class Program
+{
+    public static void RespectBrowserAcceptHeader(string[] args)
+    {
+        // <snippet_RespectBrowserAcceptHeader>
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Add services to the container.
+        builder.Services.AddControllers(options =>
+        {
+            options.RespectBrowserAcceptHeader = true;
+        });
+        // </snippet_RespectBrowserAcceptHeader>
+    }
+
+    public static void AddXmlSerializerFormatters(string[] args)
+    {
+        // <snippet_AddXmlSerializerFormatters>
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Add services to the container.
+        builder.Services.AddControllers()
+            .AddXmlSerializerFormatters();
+        // </snippet_AddXmlSerializerFormatters>
+    }
+
+    public static void JsonSerializerOptions(string[] args)
+    {
+        // <snippet_JsonSerializerOptions>
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Add services to the container.
+        builder.Services.AddControllers()
+            .AddJsonOptions(options =>
+            {
+                options.JsonSerializerOptions.PropertyNamingPolicy = null;
+            });
+        // </snippet_JsonSerializerOptions>
+    }
+
+    public static void AddNewtonsoftJson(string[] args)
+    {
+        // <snippet_AddNewtonsoftJson>
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Add services to the container.
+        builder.Services.AddControllers()
+            .AddNewtonsoftJson();
+        // </snippet_AddNewtonsoftJson>
+    }
+
+    public static void AddNewtonsoftJsonSerializerSettings(WebApplicationBuilder builder)
+    {
+        // <snippet_AddNewtonsoftJsonSerializerSettings>
+        builder.Services.AddControllers()
+            .AddNewtonsoftJson(options =>
+            {
+                options.SerializerSettings.ContractResolver = new DefaultContractResolver();
+            });
+        // </snippet_AddNewtonsoftJsonSerializerSettings>
+    }
+
+    public static void RemoveOutputFormatters(string[] args)
+    {
+        // <snippet_RemoveOutputFormatters>
+        var builder = WebApplication.CreateBuilder(args);
+
+        // Add services to the container.
+        builder.Services.AddControllers(options =>
+        {
+            // using Microsoft.AspNetCore.Mvc.Formatters;
```

Comment:
The disadvantage to this approach is you can't move the using to where it would go in standalone program.cs file. The advantage is the code is always compiled.